src/oas3/transformers/request.ts

```diff
@@ -69,11 +70,16 @@ export function translateToRequest(
     params[key].push(translateParameterObject(parameter));
   }
 
-  const body = requestBodyObject ? translateRequestBody(document, requestBodyObject) : { contents: [] };
+  let body;
+  if (isDictionary(requestBodyObject)) {
+    const resolvedRequestBodyObject = maybeResolveLocalRef(document, requestBodyObject) as RequestBodyObject;
```

Comment:
This type assertion is rather unsafe.
Unfortunately for us, `http-spec` needs to gracefully handles as many errors as possible, so we need to account for having an incorrect value.
![image](https://user-images.githubusercontent.com/9273484/114382038-2b5a5880-9b8c-11eb-95fd-8eead2ade592.png)
```suggestion
    const resolvedRequestBodyObject = maybeResolveLocalRef(document, requestBodyObject);
```

Reply:
Good point @P0lip, thank for your suggestions! I applied all of them and added an additional test to make sure it wont fail.